agent/consul/server_test.go

```diff
@@ -2107,6 +2107,77 @@ func TestServer_hcpManager(t *testing.T) {
 
 }
 
+func TestServer_addServerTLSInfo(t *testing.T) {
+	testCases := map[string]struct {
+		expectErr   bool
+		tlsConfig   tlsutil.Config
+		checkStatus func(*testing.T, hcpclient.ServerStatus)
+	}{
+		"Success": {
+			tlsConfig: tlsutil.Config{
+				InternalRPC: tlsutil.ProtocolConfig{
+					CAFile:               "../../test/ca/root.cer",
+					CertFile:             "../../test/key/ourdomain_with_intermediate.cer",
+					KeyFile:              "../../test/key/ourdomain.key",
+					VerifyIncoming:       true,
+					VerifyOutgoing:       true,
+					VerifyServerHostname: true,
+				},
+			},
+			checkStatus: func(t *testing.T, s hcpclient.ServerStatus) {
+				expected := hcpclient.ServerTLSInfo{
+					Enabled:              true,
+					CertIssuer:           "test.internal",
+					CertName:             "testco.internal",
+					CertSerial:           "40",
+					CertExpiry:           time.Date(2123, time.October, 9, 17, 20, 16, 0, time.UTC),
+					VerifyIncoming:       true,
+					VerifyOutgoing:       true,
+					VerifyServerHostname: true,
+					CertificateAuthorities: []hcpclient.CertificateMetadata{
+						{ // manual ca pem
+							CertExpiry: time.Date(2033, time.October, 30, 15, 50, 29, 0, time.UTC),
+							CertName:   "test.internal",
+							CertSerial: "191297809789001034260919865367524695178070761520",
+						},
+						{ // certificate intermediate
+							CertExpiry: time.Date(2033, time.October, 30, 15, 50, 29, 0, time.UTC),
+							CertName:   "test.internal",
+							CertSerial: "191297809789001034260919865367524695178070761520",
+						},
+					},
+				}
+
+				require.Equal(t, expected, s.ServerTLSMetadata.InternalRPC)
+
+				// TODO: remove check for status.TLS once deprecation is ready
+				require.Equal(t, expected, s.TLS)
+			},
+		},
+		"Nil Cert": {
+			checkStatus: func(t *testing.T, s hcpclient.ServerStatus) {
+				require.Empty(t, s.TLS)
+				require.Empty(t, s.ServerTLSMetadata.InternalRPC)
+			},
+		},
+		// Note on failure case: unlikely and difficult to test since
+		// NewConfigurator will fail if there are issues with the certs
```

Comment:
By failure case, I was thinking of all the paths that return an error. Let me take a pass at mocking the tls configurator.

Reply:
I took an initial pass at this [here](https://github.com/hashicorp/consul/pull/19682/commits/018cbc3af9c414ab7fe23ca8734179c3690efa3b). Would you mind taking a look for initial feedback? From a quick look, I didn't notice an automated way to generated mocks but could have missed it. I created an interface and mock of a subset of methods that were used since there were many (~29). 🤔 it's quite a bit of extra code to achieve full coverage. The error cases are a bit unlikely so open to input on balance of coverage vs. maintainability.